src/executableserializer.cpp

```diff
@@ -0,0 +1,197 @@
+        {
+            printf("Loading sections info...\n");
+        }
+
+        const auto &sections = js.at(EXE_SECTIONS_SECTION);
+        for (const auto &section : sections)
+        {
+            std::string name = section.at("name").get<std::string>();
+            if (name.empty())
+                continue;
+
+            ExeSectionInfo *sectionInfo = exe.find_section(name);
+            if (!sectionInfo)
+            {
+                if (m_verbose)
+                {
+                    printf("Section '%s' not found in binary\n", name.c_str());
+                }
+                continue;
+            }
+
+            std::string type = section.at("type").get<std::string>();
+            sectionInfo->type = to_section_type(type.c_str());
+
+            if (section.contains("address"))
+                section.at("address").get_to(sectionInfo->address);
+            if (section.contains("size"))
+                section.at("size").get_to(sectionInfo->size);
+        }
+    }
+
+    if (js.contains(EXE_OBJECTS_SECTION))
+    {
+        if (m_verbose)
+        {
+            printf("Loading objects section...\n");
+        }
+
+        const auto &objects = js.at(EXE_OBJECTS_SECTION);
+        for (const auto &object : objects)
+        {
+            ExeObject obj;
+            object.at("name").get_to(obj.name);
+            const auto &sections = object.at("sections");
+            for (const auto &section : sections)
+            {
+                ExeObjectSection objSection;
+                section.at("name").get_to(objSection.name);
+                section.at("offset").get_to(objSection.offset);
+                section.at("size").get_to(objSection.size);
+                obj.sections.push_back(objSection);
+            }
+            exe.m_targetObjects.push_back(obj);
+        }
+    }
+}
+
+void ExecutableSerializer::save_json(nlohmann::json &js, const Executable &exe) const
+{
+    if (m_verbose)
+    {
+        printf("Saving config section...\n");
+    }
+    // Config section
+    js[EXE_CONFIG_SECTION] = {
+        {"codealign", exe.m_imageData.codeAlignment},
+        {"dataalign", exe.m_imageData.dataAlignment},
+        {"codepadding", exe.m_imageData.codePad},
+        {"datapadding", exe.m_imageData.dataPad}};
+
+    if (m_verbose)
+    {
+        printf("Saving symbols section...\n");
+    }
+    // Symbols section
+    auto &symbols = js[EXE_SYMBOLS_SECTION] = nlohmann::json::array();
+    for (const auto &symbol : exe.m_symbols)
+    {
+        symbols.push_back({{"name", symbol.name}, {"address", symbol.address}, {"size", symbol.size}});
+    }
+
+    // Sections section
+    auto &sections = js[EXE_SECTIONS_SECTION] = nlohmann::json::array();
+    for (const auto &section : exe.m_sections)
+    {
+        sections.push_back(
+            {{"name", section.name},
+             {"type", to_string(section.type)},
+             {"address", section.address},
+             {"size", section.size}});
```

Comment:
Can you try to rewrite all these json read and write blocks to be similar to how pdbreadertypes does it:

```cpp
void to_json(nlohmann::json &js, const PdbFunctionInfo &d)
{
    js = nlohmann::json{
        {"file_id", d.sourceFileId},
        {"compiland_id", d.compilandId},
        {"address", d.address},
        {"debug_start_address", d.debugStartAddress},
        {"debug_end_address", d.debugEndAddress},
        {"len", d.length},
        {"call", d.call},
        {"name_decorated", d.decoratedName},
        {"name_undecorated", d.undecoratedName},
        {"name_global", d.globalName},
        {"lines", d.sourceLines},
    };
}

void from_json(const nlohmann::json &js, PdbFunctionInfo &d)
{
    js.at("file_id").get_to(d.sourceFileId);
    js.at("compiland_id").get_to(d.compilandId);
    js.at("address").get_to(d.address);
    js.at("debug_start_address").get_to(d.debugStartAddress);
    js.at("debug_end_address").get_to(d.debugEndAddress);
    js.at("len").get_to(d.length);
    js.at("call").get_to(d.call);
    js.at("name_decorated").get_to(d.decoratedName);
    js.at("name_undecorated").get_to(d.undecoratedName);
    js.at("name_global").get_to(d.globalName);
    js.at("lines").get_to(d.sourceLines);
}
```

This likely makes it easier to maintain, but perhaps gives less flexibility in how the data is laid out in json.

Reply:
Ok done, and I match the style/structure of pdbreader, now we have executable and executabletypes. I tried to make function names and things consistent.